### PR TITLE
Add optional name for runnable sequence

### PR DESCRIPTION
**Pull request by nfcampos**
Fixes # (issue)

### CODE REVIEW

File: langchain-core/src/runnables/base.ts [hunk 1]
```diff
@@ -124,6 +124,15 @@ export abstract class Runnable<
 {
   protected lc_runnable = true;
 
+  name?: string;
+
+  getName(suffix?: string): string {
```
Comment (jacoblee93) [inline langchain-core/src/runnables/base.ts:129]: Do we need to add to the interface?

File: langchain-core/src/runnables/base.ts [hunk 2]
```diff
@@ -124,6 +124,15 @@ export abstract class Runnable<
...
+    const name =
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      this.name ?? (this.constructor as any).lc_name ?? this.constructor.name;
```
Comment (jacoblee93) [inline langchain-core/src/runnables/base.ts:132]: `lc_name` is a function

Comment (nfcampos) [reply langchain-core/src/runnables/base.ts]: Oops

File: langchain-core/src/runnables/base.ts (hunk 1 above)
Comment (nfcampos) [reply langchain-core/src/runnables/base.ts:129]: Yes this will be used by the schemas too, and I think it makes sense to be public

Comment (nfcampos) [reply langchain-core/src/runnables/base.ts:129]: Actually I won't add to interface

**Pull request merged**